lib/src/api/opt/query.rs

```diff
@@ -322,9 +336,15 @@ where
 {
 	fn query_result(self, QueryResponse(map): &mut QueryResponse) -> Result<Vec<T>> {
 		let (index, key) = self;
-		let response = match map.get_mut(&index) {
+		let mut response = match map.get_mut(&index) {
```

Comment:
```suggestion
		let response = match map.get_mut(&index) {
```

Maybe I'm missing something but I don't think the mutable reference itself needs to be mutable here.

Reply:
We need to iterate over it a few lines below it

Reply:
Yes but it's already a mutable reference and we are using `iter_mut` so we should be good, right? The reference is already mutable. The variable itself doesn't need to be because of that.

Reply:
<img width="1236" alt="image" src="https://github.com/surrealdb/surrealdb/assets/17960250/22140ea8-d971-4d4b-9adc-f678fc8c6955">

Reply:
Ah, I see! That match statement is now returning a `Vec<Value>`. That's what I was missing. Previously it was returning a `&mut Vec<Value>`.